### PR TITLE
[PIPE-408] Award search gold

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Adds options to load the award gold table

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [X] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [PIPE-408](https://federal-spending-transparency.atlassian.net/browse/PIPE-408):
    - [X] Link to this Pull-Request
    - [X] Performance evaluation of affected (API | Script | Download)
    - [X] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/search/delta_models/award_search.py [hunk 1]
```diff
@@ -1,126 +1,131 @@
...
-    "officer_2_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_2_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_3_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_3_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_4_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_4_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_5_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_5_name": {"delta": "STRING", "postgres": "TEXT"},
+    "treasury_account_identifiers": {"delta": "ARRAY<INTEGER>", "postgres": "INTEGER[]", "gold": False},
+    "award_id": {"delta": "LONG NOT NULL", "postgres": "BIGINT NOT NULL", "gold": True},
```
Comment (collinwr) [inline usaspending_api/search/delta_models/award_search.py:5]: `award_id` was on the original table, so `gold` should be `False` here right?

Comment (collinwr) [reply usaspending_api/search/delta_models/award_search.py:5]: Same with a few fields below such as `category`, `type`, `type_description`, etc

File: usaspending_api/search/delta_models/award_search.py [hunk 2]
```diff
@@ -1,126 +1,131 @@
...
+    "recipient_unique_id": {"delta": "STRING", "postgres": "TEXT", "gold": False},
+    "parent_recipient_unique_id": {"delta": "STRING", "postgres": "TEXT", "gold": False},
+    "recipient_uei": {"delta": "STRING", "postgres": "TEXT", "gold": False},
+    "parent_uei": {"delta": "STRING", "postgres": "TEXT", "gold": False},
+    "business_categories": {"delta": "ARRAY<STRING>", "postgres": "TEXT[]", "gold": False},
+    "total_subaward_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)", "gold": True},
+    "subaward_count": {"delta": "INTEGER", "postgres": "INTEGER", "gold": True},
+    "action_date": {"delta": "DATE", "postgres": "DATE", "gold": False},
+    "fiscal_year": {"delta": "INTEGER", "postgres": "INTEGER", "gold": False},
+    "last_modified_date": {"delta": "DATE", "postgres": "DATE", "gold": True},
```
Comment (collinwr) [inline usaspending_api/search/delta_models/award_search.py:46]: I believe this field was on the original model

Comment (emily-brents) [reply usaspending_api/search/delta_models/award_search.py]: fixed

File: usaspending_api/search/delta_models/award_search.py [hunk 3]
```diff
@@ -1,126 +1,131 @@
...
-    "officer_2_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_3_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_3_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_4_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_4_name": {"delta": "STRING", "postgres": "TEXT"},
-    "officer_5_amount": {"delta": "NUMERIC(23, 2)", "postgres": "NUMERIC(23, 2)"},
-    "officer_5_name": {"delta": "STRING", "postgres": "TEXT"},
+    "treasury_account_identifiers": {"delta": "ARRAY<INTEGER>", "postgres": "INTEGER[]", "gold": False},
+    "award_id": {"delta": "LONG NOT NULL", "postgres": "BIGINT NOT NULL", "gold": False},
+    "transaction_unique_id": {"delta": "STRING", "postgres": "TEXT", "gold": False},
```
[the remark this answers is not in the document]
Comment (emily-brents) [reply usaspending_api/search/delta_models/award_search.py]: fixed

**Review by collinwr — APPROVED** [review]

**Pull request merged**